instrumentation/resources/library/src/test/java/io/opentelemetry/instrumentation/resources/JarServiceNameProviderTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.resources;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import java.nio.file.Path;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class JarServiceNameProviderTest {
+
+  @Mock ConfigProperties config;
+
+  @Test
+  void createResource_empty() {
+    JarServiceNameProvider serviceNameProvider =
+        new JarServiceNameProvider(() -> new String[0], prop -> null, file -> false);
+
+    Resource resource = serviceNameProvider.createResource(config);
+
+    assertThat(resource.getAttributes()).isEmpty();
+  }
+
+  @Test
+  void createResource_noJarFileInArgs() {
+    String[] args = new String[] {"-Dtest=42", "-Xmx666m", "-jar"};
+    JarServiceNameProvider serviceNameProvider =
+        new JarServiceNameProvider(() -> args, prop -> null, file -> false);
+
+    Resource resource = serviceNameProvider.createResource(config);
+
+    assertThat(resource.getAttributes()).isEmpty();
+  }
+
+  @Test
+  void createResource_processHandleJar() {
+    String[] args =
+        new String[] {"-Dtest=42", "-Xmx666m", "-jar", "/path/to/app/my-service.jar", "abc", "def"};
+    JarServiceNameProvider serviceNameProvider =
+        new JarServiceNameProvider(() -> args, prop -> null, file -> false);
```

Comment:
This gave me a little bit of pause because the "does the file exist" check is returning false here....and the test passes because it's not used. Definitely prefer passing `null` for tests where the argument is not used (there are a few others too I believe).

Reply:
How about `() -> throw new AssertionError()` instead? Passing null results in null warnings

Reply:
We should allow passing null in tests!! I think that suggestion is an improvement, though.